text/0000-wheel-platform-support.md

```diff
@@ -0,0 +1,217 @@
+- Feature Name: "Wheel" support in RubyGems
+- Start Date: 2025-08-20
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Add wheel-format platform support to RubyGems for more fine-grained matching for binary gem distribution through ruby/api tags and platform tags (`whl-{abi_tags}-{platform_tags}`). This enables "platform matching" based on Ruby ABI version, os, os version, cpu architecture, and libc implementation (for generic linux platforms).
+
+# Motivation
+
+Binary gem distribution currently uses coarse-grained platform matching (e.g., "x86_64-linux") that cannot distinguish between:
+
+- Different Ruby language and ABI versions (Ruby 3.2 vs 3.3 native extension binaries are incompatible)
+- Linux libc implementation versions (musl and glibc can introduce new symbols in new versions)
+
+This leads to native extension gems either failing at runtime, requiring compilation from source even when compatible binaries exist, or packaging _multiple_ extension binaries in a single gem (of which each installation will use only one). The wheel format provides a solution [inspired by the the Python ecosystem](https://packaging.python.org/en/latest/specifications/platform-compatibility-tags/) that addresses these issues.
+
+# Guide-level explanation
+
+When installing gems with native extensions, RubyGems will now automatically select the most appropriate pre-compiled binary based on your exact Ruby version and platform:
+
+```bash
+$ gem install nokogiri
+# On Ruby 3.3, x86_64 Linux with glibc 2.28:
+# Installs nokogiri-1.16.0-whl-rb33-manylinux_2_28_x86_64.gem (pre-compiled)
+
+# On Ruby 3.2, same platform:
+# Falls back to nokogiri-1.16.0.gem (compiles from source)
+```
+
+Gem authors can build platform-specific gems:
+
+```sh
+gem build --platform "whl-rb33-x86_64_linux"  # Specific to Ruby 3.3 on Linux x86_64
+```
+
+
+The wheel platform format follows the pattern `whl-{abi_tags}-{platform_tags}`:
+
+- `abi_tags`: Ruby ABI version (e.g., `rb33` for Ruby 3.3, `any` for pure Ruby, `cr3` for any CRuby with major version 3)
```

Comment:
Is there a different meaning between `Ruby` (from  `rb33` for Ruby 3.3) and `CRuby` (from `cr3` for any CRuby with major version 3) on this line?
I think in both cases you mean the Ruby C API as implemented by CRuby X.Y, IOW the CRuby ABI.

Reply:
The ABI version is AFAIK always RUBY_ENGINE-specific.
In theory a new Ruby implementation could try to support the exact same ABI as CRuby (there is none currently), but if so then RubyGems should just consider e.g. `cr33` to be compatible with that new Ruby implementation.

IOW, I think this should be removed:
```
`rb33` for Ruby 3.3,
```
Because there is no such thing as a "Ruby" ABI 3.3 which is compatible with more than CRuby ABI 3.3.